examples/examples.rs

```diff
@@ -82,37 +80,26 @@ fn examples() -> Result<()> {
     // This API handles variadics using Heterogeneous Lists.  This is one way to
     // call a function with multiple parameters:
 
-    print.call::<_, ()>(
-        hlist!["hello", "again", "from", "rust"],
-    )?;
+    print.call::<_, ()>(("hello", "again", "from", "rust"))?;
 
     // You can bind rust functions to lua as well
 
-    let check_equal = lua.create_function(|lua, args| {
-        // Functions wrapped in lua receive their arguments packed together as
-        // MultiValue.  The first thing that most wrapped functions will do
-        // is "unpack" this MultiValue into its parts.  Due to lifetime type
-        // signature limitations, this cannot be done automatically from the
-        // function signature, but this will be fixed with ATCs.  Notice the use
-        // of the hlist macros again.
-        let hlist_pat![list1, list2] = lua.unpack::<HList![Vec<String>, Vec<String>]>(args)?;
-
+    let check_equal = lua.create_function(|_, (list1, list2): (Vec<String>, Vec<String>)| {
```

Comment:
> No no, that's perfectly fine. I was referring to "make sure that the top-level Lua object has a stable address", which would not fix the issue, which is basically arbitrary unsafe lifetime transmutation.

Okay, then I think we're in agreement.  I feel like you know more about these lifetime problems than I do so I'm trying to make sure I understand what you're saying.

> It might be possible to express the "shortest borrowed lifetime" as a lifetime parameter of Lua, which should solve this.

So, you could make Lua itself take a lifetime parameter, and all closures would have to outlive the lifetime parameter?  It would just be PhantomData in Lua?  How do you get the lifetime into the callback, the callbacks can't know which Lua they're ultimately being called from except by getting the main state pointer.  I might be misunderstanding what you're saying though.

> My experiments indicate that a closure borrowing the outside lua will not compile. In particular, this works, but not this, and I don't see a way to make the second version work. Therefore, the parameter needs to be kept.

Which parameter needs to be kept, I'm not following.